specification/hugr.md

```diff
@@ -1757,6 +1757,27 @@ allocation for all `Qubit` wires.
 If further the program does not contain any `qalloc` or `qfree`
 operations we can state the program only uses `N` qubits.
 
+#### Angles
+
+The Quantum extension also defines a specialized `angle<N>` type which is used
+to express parameters of rotation gates. The type is parametrized by the
+_log-denominator_, which is an integer $N \in [0, 53]$; angles with
+log-denominator $N$ are multiples of $2 \pi / 2^N$, where the multiplier is an
+unsigned `int<N>`. The maximum log-denominator $53$ effectively gives the
+resolution of a `float64` value; but note that unlike `float64` all angle values
+are equatable and hashable.
```

Comment:
I'm wondering whether `x / (2pi/2^N)` has the same hash as `(2x)/(2pi/2^(N+1))`, but I don't think you give the hash "algorithm" so fine

Reply:
If we restrict the numerator to the range $[0, 2^n]$, which I think we should, then all values with a given denominator $2^n$ have different hashes. Interesting question is whether values with different denominators should have the same hash if they represent the same angle: e.g. $1 \times 2\pi/2^2$ and $2 \times 2\pi/2^3$. I think probably not, since they are different types and you can do different things with them.